contracts/coordinator/src/contract/execute.rs

```diff
@@ -307,14 +309,24 @@ pub fn register_deployment(
         state::protocol_contracts(deps.storage).change_context(Error::ProtocolNotRegistered)?;
 
     let router = router_api::client::Router::new(protocol_contracts.router);
+    let coordinator: Client =
+        client::ContractClient::new(deps.querier, &env.contract.address).into();
```

Comment:
why does the coordinator need a client to call itself? just call `register_chain` directly 🙂